public/request/user/update-email.php

```diff
@@ -15,25 +16,31 @@
 
 if ($email !== $email2) {
     header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_notmatch");
-} else {
-    if (filter_var($email, FILTER_VALIDATE_EMAIL) == false) {
-        header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_badnewemail");
-    } else {
-        if (authenticateFromCookie($user, $permissions, $userDetail)) {
-            $query = "UPDATE UserAccounts SET EmailAddress='$email', Permissions=0, Updated=NOW() WHERE User='$user'";
-            $dbResult = s_mysql_query($query);
-            if ($dbResult) {
-                sendValidationEmail($user, $email);
-
-                addArticleComment('Server', ArticleType::UserModeration, $userDetail['ID'],
-                    $user . ' changed their email address');
-
-                header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_changeok");
-            } else {
-                header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_generalerror");
-            }
-        } else {
-            header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_badcredentials");
-        }
+    exit;
+}
+if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+    header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_badnewemail");
+    exit;
+}
+
+if (authenticateFromCookie($user, $permissions, $userDetail)) {
+    $dbResult = s_mysql_query(
+        "UPDATE UserAccounts SET EmailAddress='$email', Permissions=" . Permissions::Unregistered . ", Updated=NOW() WHERE User='$user'"
+    );
+
+    if (!$dbResult) {
+        header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_generalerror");
+        exit;
     }
+
+    sendValidationEmail($user, $email);
+
+    addArticleComment('Server', ArticleType::UserModeration, $userDetail['ID'],
+        $user . ' changed their email address'
+    );
+
+    header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_changeok");
+    exit;
 }
+
+header("Location: " . getenv('APP_URL') . "/controlpanel.php?e=e_badcredentials");
```

Comment:
Since you're already flattening the if chain, why not put this in a `!authenticateFromCookie` and leave the main functional part unindented?

Reply:
Missed that opportunity :) Thanks!